### DISCUSSION TITLE
Open discussion - Do we want to drill down HTML props to sub components by default ?

### DISCUSSION THREAD

**Discussion by lucasbordeau** [twentyhq/twenty#840] [Engineering vision]

**Comment by lucasbordeau** [comment]
As an interesting tradeoff, we can use the className prop as described in https://styled-components.com/docs/basics#styling-any-component, so any component we expose can be styled.

If we need to style a sub-component, we can just pass className to this sub-component. 

I think it's better to keep our components closed to the props we specify, allowing to pass all of a default tag props by default to every component we have would open everything too much. 

That would be understandable for a component library like Material-UI but not really for a business application like Twenty, where we want to compose UIs fast with a coherent and simple design system.

**Reply by charlesBochet** [reply]
I agree that we want to restrict the API as we want to maintain an interface on the long term. If developers start using our components in a way we don't know about, we won't be able to make the components evolve in the future. By enforcing a restricted API, we make an agreement with our components users that this restricted API is how to use our component.

Regarding className, and StyledComponent, I think it should only be used to introduce styling that depends on the external context of the component: margin, alignment.

**Reply by lucasbordeau** [reply]
Then we could say that className is not put by default because in the vast majority of cases we will use flex-box and container to align components, as it is the standard way to do.
